### PR TITLE
Add file appender.  Numerous fixes to logging.

### PR DESCRIPTION
**Pull request by brianbegygr**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,13 @@
+## v17.3.2.0 / 2017 Oct 06
+
+* **Fix**  -  Numerous fixes to logging.  
+* **Add** Expose Shutdown() on logger.  Handle recursive errors.
```
Comment (jeffpsherman) [inline CHANGELOG.md]: Inconsistent formatting `* **Add** ` and `* **Add** - `

File: GuaranteedRate.Sextant/Logging/File/FileLogAppender.cs [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+                            }
+                        }
+                        catch (System.IO.IOException iox)
+                        {
+                            return false;
+                        }
+                        catch (Exception ex)
+                        {
+
+                            throw;
```
Comment (jeffpsherman) [inline GuaranteedRate.Sextant/Logging/File/FileLogAppender.cs]: Why catch and throw?
What's the value of not catching if you're just going to throw?

File: GuaranteedRate.Sextant/Logging/Logger.cs [hunk 3]
```diff
@@ -2,13 +2,20 @@
...
+using System.Linq;
+using System.Threading.Tasks;
+using Belikov.GenuineChannels;
 using GuaranteedRate.Sextant.Config;
+using GuaranteedRate.Sextant.Logging.Console;
 using GuaranteedRate.Sextant.Logging.Elasticsearch;
+using GuaranteedRate.Sextant.Logging.File;
 using GuaranteedRate.Sextant.Logging.Loggly;
+using GuaranteedRate.Sextant.Metrics;
+using Org.BouncyCastle.Asn1.X509.Qualified;
```
Comment (jeffpsherman) [inline GuaranteedRate.Sextant/Logging/Logger.cs]: Is this legit?

File: GuaranteedRate.Sextant/WebClients/AsyncEventReporter.cs [hunk 4]
```diff
@@ -91,59 +84,85 @@ private void Init()
...
+                    catch (Exception ex)
+                    {
+
+                            Logger.Warn(Name, $"Exception processing reporter queue:{ex}", excludedReporters);
+
+                        throw;
+                    }
+
                 }
+                Console.WriteLine("####----finished.");
```
Comment (jeffpsherman) [inline GuaranteedRate.Sextant/WebClients/AsyncEventReporter.cs]: That's going to do interesting things when running without a console

File: GuaranteedRate.Sextant/WebClients/AsyncEventReporter.cs [hunk 5]
```diff
@@ -160,79 +179,38 @@ protected virtual void ExtraSetup(WebRequest webRequest)
...
                 {
-                    Shutdown();
+                    Shutdown(_shutdownTimeout);
                 }
             }
 
             disposedValue = true;
         }
 
+#pragma warning disable CC0029 // Disposables Should Call Suppress Finalize
```
Comment (jeffpsherman) [inline GuaranteedRate.Sextant/WebClients/AsyncEventReporter.cs]: Resharper artifact?

File: GuaranteedRate.Sextant/WebClients/AsyncEventReporter.cs [hunk 6]
```diff
@@ -160,79 +179,38 @@ protected virtual void ExtraSetup(WebRequest webRequest)
...
         public void Dispose()
+#pragma warning restore CC0029 // Disposables Should Call Suppress Finalize
```
Comment (jeffpsherman) [inline GuaranteedRate.Sextant/WebClients/AsyncEventReporter.cs]: Resharper artifact?

File: GuaranteedRate.Sextant/WebClients/SecureAsyncEventReporter.cs [hunk 7]
```diff
@@ -5,7 +5,7 @@
 
 namespace GuaranteedRate.Sextant.WebClients
 {
-    public class SecureAsyncEventReporter : AsyncEventReporter
+    public class SecureAsyncEventReporter : AsyncWebEventReporter
```
Comment (jeffpsherman) [inline GuaranteedRate.Sextant/WebClients/SecureAsyncEventReporter.cs]: Are there any other event reporter implementations in our git repos that will need to be updated by this refactor?

File: GuaranteedRate.Sextant/WebClients/AsyncEventReporter.cs (hunk 4 above)
Comment (brianbegygr) [reply GuaranteedRate.Sextant/WebClients/AsyncEventReporter.cs]: gah.  Missed that one.

File: GuaranteedRate.Sextant/Logging/Logger.cs (hunk 3 above)
Comment (brianbegygr) [reply GuaranteedRate.Sextant/Logging/Logger.cs]: unused.  Must be resharper trying to help.

File: GuaranteedRate.Sextant/Logging/File/FileLogAppender.cs (hunk 2 above)
Comment (brianbegygr) [reply GuaranteedRate.Sextant/Logging/File/FileLogAppender.cs]: Only value is when debugging.  Removing.

File: GuaranteedRate.Sextant/WebClients/AsyncEventReporter.cs (hunk 6 above)
Comment (brianbegygr) [reply GuaranteedRate.Sextant/WebClients/AsyncEventReporter.cs]: You need that or it does a warn.

File: GuaranteedRate.Sextant/WebClients/SecureAsyncEventReporter.cs (hunk 7 above)
Comment (brianbegygr) [reply GuaranteedRate.Sextant/WebClients/SecureAsyncEventReporter.cs]: Looks like Kerosene and Dreck.  I'm open to keeping AsyncEventReporter as a web reporter and make something like AsyncLocalEventReporter and explain in a comment.  Or we can just break compatibility and let the chips fall.

**Pull request merged**